src/ssl.mli

```diff
@@ -523,3 +514,40 @@ val input_int : socket -> int
 
 (** Write an integer on an SSL socket. *)
 val output_int : socket -> int -> unit
+
+(** In `SslNoRelease`, you will find equivalent of the previous function that
+    do not release the OCaml runtime lock before calling the underlying SSL
+    primitives. This should be preferer for non blocking sockets where the
+    usual semantics apply, i.e. handling of `EWOULDBLOCK`, `EGAIN`, etc.
+    In could also be used with blocking socket if you program has only one thread.
+
+    This saves some buffer copy and can be up to 2x faster.
+*)
+module SslNoRelease : sig
```

Comment:
This should be called `SslBlocking` or `Ssl_blocking`

Reply:
> This should be called `SslBlocking` or `Ssl_blocking`

I disagree: SslNoRelease has at least two use cases with Blocking socket
- Mono threaded code (like a simple client)
- fork/mmap based parallelism (will probably disappear with 5.0 ?), which is mono threaded too.

Still I am not fully happy with the name.

Reply:
> This should be called `SslBlocking` or `Ssl_blocking`

And more importantly, these are Blocking only if the socket is configured as blocking, so the name is misleading

a better name could be SslForNonBloackingSocket ?

Reply:
I didn't read "blocking" as "socket in blocking mode", but rather "blocking" because it doesn't release the OCaml runtime (so it can't be "async". I was using "blocking" as the opposite of "async"). maybe "sync" is a better name then?

Reply:
Blocking was the old terminology in ocaml c APIs and it was super confusing. The new terminology explicitly mentioned release/no release, i think it's generally much better.

Reply:
It is the same SslSync is misleading, because it only describes the semantic you get with Blocking socket, while the module SslNoRelease can be used for Async semantics with non-blocking socket, which, I think, is its common use case. 

Naming by the semantics will not work because the semantics depends upon the socket type. This is why I ressorted to naming by what it does. Here is a list of proposition that could help us find a good name:

- SslNoRelease
- SslKeepLock
- SslForNonBlock (exclude sync mono threaded use case)
- SslDirect (also express that we avoid a buffer and that it is faster)

I like SslDirect: it is short and a fair description. You can not guess (wrongly) its meaning without looking at the documentation. Which is what we want for such an advanced feature ?

Reply:
asked a friend for some more suggestions:
<img width="905" alt="image" src="https://github.com/savonet/ocaml-ssl/assets/661909/4605e0fe-c744-4648-ad91-0ed6ce22df49">

Reply:
how about `Ssl.Runtime_locked` for the new set of functions? I think that's clear enough.

Reply:
Yes, I like Ssl.RuntimeLocked (I prefer underscore for lowercase ident and capital for upper case ident in OCaml). Super :

Reply:
[856a5c5](https://github.com/savonet/ocaml-ssl/pull/106/commits/856a5c57ce08cc1878acadc5f1409eb87d19eb19)
introduces the new name!

Reply:
I realize it's your personal preference to use camel-casing, but there are no camel-cased names in this library, so we should probably preserve that property to minimize the surprise factor.

Reply:
You are clearly right. 

[b5c6966](https://github.com/savonet/ocaml-ssl/pull/106/commits/b5c6966657735806ac6096fd349c294a3ce2dd76) solves this